src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h

```diff
@@ -568,6 +606,64 @@ namespace OpenMS
       const bool is_left_half
     ) const;
 
+    /**
+      @brief Given a saturated or cutoff peak, extract a training set to be used
+      with the gradient descent algorithm.
+
+      The algorithm tries to select a subset of points that best describe a non-saturated
+      peak. The decision of which points to skip is based on the derivatives between consecutive points.
+
+      It first selects all those points whose intensity is below a certain value (`intensity_threshold`).
+      Then, the derivatives of all the remaining points are computed. Based on the results,
+      the algorithm selects those points that present a high derivative.
+      Once a low value is found, the algorithm stops taking points from that side.
+      It then repeats the same procedure on the other side of the peak.
+      The skipped points will be found in the middle part of the peak, which is the saturated part.
```

Comment:
Will skipped points always be found?  I.e., if we give it a perfect looking peak will it use all points or will it not use all points for the fitting procedure?  Also, comments seem quite geared towards saturated peaks.  Is the use of changing derivatives also able to deal with cutoff peaks?  From the data I have seen so far, that seems to be the case.

Reply:
Case of a well shaped peak:
In this case, it might happen that, if the points toward the peak's centre have similar intensity, then the points on the middle might be skipped. That is why I chose a low value (`0.4`) to compute `derivative_threshold`. Looking at the peaks I've seen, it looks like no points were skipped for well shaped peaks. But I guess it could happen. Anyway it wouldn't negatively affect the final shape, at least not too much, given the "conservative" behavior of the method (I mean, it tries to keep as much information as possible from the input peak, discarding points only when the given point really wouldn't help in the estimation).

The method also supports cutoff peaks. What happens is that probably no points will be found below (0.8 * max element) (or maybe just a very low number of them), and then it will work on the derivatives.